src/ert/scheduler/scheduler.py

```diff
@@ -107,8 +115,17 @@ def kill_all_jobs(self) -> None:
 
     async def cancel_all_jobs(self) -> None:
         self._cancelled = True
+        await self._cancel_tasks()
+
+    async def _cancel_tasks(self) -> None:
         for task in self._tasks.values():
             task.cancel()
+        with suppress(asyncio.TimeoutError):
+            await asyncio.wait(
+                self._tasks.values(),
+                timeout=1.0,
+                return_when=asyncio.ALL_COMPLETED,
+            )
```

Comment:
Here, `_cancel_tasks(...)` is a very general method, and could be fine with just taking in a list of tasks to cancel instead of the class instance. Could we either make this a general function outside the class, or give it a different name? I think `_cancel_tasks` should be called `_cancel_all_jobs` if we keep it as a method.

Reply:
I'd change it to `_cancel_all_jobs` since currently we don't need to generalize this function but if there is a need to we can create one :)